From c4165b1928607c06705cd4576629a68089dac798 Mon Sep 17 00:00:00 2001
From: minoplhy <c@3qx.nl>
Date: Sat, 21 Feb 2026 15:18:06 +0700
Subject: [PATCH] Downgrade Go code format to below 1.26.0

Go 1.26 has update some of its code format that made it not compatible with Go version 1.25.5 .

This patch is based on commit : https://github.com/henrygd/beszel/commit/04600d83cc01b1c009f4f8dccb60d4c7754a7bde
---
 agent/smart.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/agent/smart.go b/agent/smart.go
index 2a779589..6bffea07 100644
--- a/agent/smart.go
+++ b/agent/smart.go
@@ -476,7 +476,7 @@ func (sm *SmartManager) CollectSmart(deviceInfo *DeviceInfo) error {
 	output, err := cmd.CombinedOutput()
 
 	// Check if device is in standby (exit status 2)
-	if exitErr, ok := errors.AsType[*exec.ExitError](err); ok && exitErr.ExitCode() == 2 {
+	if exitErr, ok := err.(*exec.ExitError); ok && exitErr.ExitCode() == 2 {
 		if hasExistingData {
 			// Device is in standby and we have cached data, keep using cache
 			return nil
-- 
2.53.0

